packages/client/src/runtime/core/request/createRequestProxy.ts

```diff
@@ -0,0 +1,43 @@
+import { defaultProxyHandlers } from '../utils/defaultProxyHandlers'
+import { RequestPromiseOptions } from './RequestPromise'
+
+export type RequestProxyOptions = RequestPromiseOptions & {
+  forbiddenProps?: string[]
+}
+
+/**
+ * Proxy that takes over the client promises to pass `options`
+ * @param thing to be proxied
+ * @param options to be passed down to the promise
+ * @returns
+ */
+export function createRequestProxy<T>(thing: T, proxyOptions?: RequestProxyOptions): T {
+  // we only wrap within a proxy if it's possible: if it's an object
+  if (thing && typeof thing !== 'object') return thing
+
+  return new Proxy(thing as object, {
+    get: (target, prop) => {
+      // we don't want to allow any calls to our `forbidden` methods
+      if (proxyOptions?.forbiddenProps?.includes(prop as string)) return undefined
+
+      // we override and handle every function call within the proxy
+      if (typeof target[prop] === 'function') {
+        return (...args: unknown[]) => {
+          // we hijack promise calls to pass txId to prisma promises
+          if (prop === 'then') return target[prop](args[0], args[1], proxyOptions)
+          if (prop === 'catch') return target[prop](args[0], proxyOptions)
+          if (prop === 'finally') return target[prop](args[0], proxyOptions)
+          if (prop === '$extends') return target[prop](args[0], proxyOptions)
```

Comment:
Shouldn't `$extends` be among forbidden properties? Or am I misunderstanding something? Like, you should not be able to do:

```ts
prisma.$transaction((tx) => {
   tx.extends(someExtension)
})
```

Reply:
I think that should work, why should you not be able to do that?

Reply:
Nothing really wrong, it's just seems weird to me. Like, itx ideally live for couple of dozens milliseconds, it seems weird to add extension that would be active just for that duration.

Reply:
True. On the other hand, extensions should generally have minimal instantiation overhead. I think extending queries could be useful there, and perhaps results too. For sure, model and client would not make much sense there.

Reply:
I've restricted it. Well see if users ever ask for it.